### PR TITLE
aichat 0.8.0 (new formula)

### PR DESCRIPTION
**Pull request by sigoden**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----
Aichat is a powerful chatgpt cli.
Aichat's home site is https://github.com/sigoden/aichat.

`brew audit --new aichat` fails with:

```
aichat:
  * GitHub repository too new (<30 days old)
Error: 1 problem in 1 formula detected
```

This is because chatgpt is quite popular, this repo already got 564 stars and 34 forks at time of request.

As for testing, I just added a check for the --version output.  All other functionality would need a api_key to interact with gpt.

### CODE REVIEW

File: Formula/aichat.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    version_out = shell_output("#{bin}/aichat --version")
+    assert_match "aichat #{version}", version_out
```
Comment (SMillerDev) [inline Formula/aichat.rb:17]: We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

**Comment by sigoden** [comment]
@SMillerDev I have refactored the test.

File: Formula/aichat.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+class Aichat < Formula
+  desc "ChatGPT cli"
```
Comment (dawidd6) [inline Formula/aichat.rb:2]:
```suggestion
  desc "ChatGPT/GPT-3.5/GPT-4 in the terminal"
```

I would rephrase the description like that. Whatcha think?

Comment (sigoden) [reply Formula/aichat.rb:2]: What if OpenAI releases more models in the future? Do we need to change every time a new model comes out?

**Review by dawidd6 — APPROVED** [review]

File: Formula/aichat.rb (hunk 2 above)
Comment (chenrui333) [reply Formula/aichat.rb:2]: yeah, I think `ChatGPT cli` make sense, they probably gonna release a new version next year. :)

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@sigoden, thanks for your contribution to Homebrew! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!
